src/seismometer/controls/explore.py

```diff
@@ -0,0 +1,1109 @@
+        return self.model_options.intervention
+
+    @property
+    def reference_time(self) -> str:
+        """reference time column descriptor"""
+        return self.model_options.reference_time
+
+
+class ExplorationCohortInterventionEvaluationWidget(ExlorationWidget):
+    """
+    A widget for exploring the model performance based on the subgroups of a cohort column.
+    """
+
+    def __init__(
+        self,
+        title: str,
+        cohort_groups: dict[str, tuple[Any]],
+        outcome_names: tuple[Any],
+        intervention_names: tuple[Any],
+        reference_time_names: tuple[Any],
+    ):
+        """
+        Exploration widget for plotting of interventions and outcomes accross categories in a cohort group.
+
+        Parameters
+        ----------
+        title : str
+            title of the control
+        cohort_groups : dict[str, tuple[Any]]
+            cohort names and category values
+        outcome_names : tuple[Any], optional
+            outcome descriptors, by default None
+        intervention_names : tuple[Any], optional
+            intervention descriptors, by default None
+        reference_time_names : tuple[Any], optional
+            reference time descriptors, by default None
+        """
+        super().__init__(
+            title,
+            option_widget=ModelInterventionAndCohortGroupWidget(
+                cohort_groups,
+                outcome_names,
+                intervention_names,
+                reference_time_names,
+            ),
+        )
+
+
+@export
+class ExploreCohortInterventionTimes(ExplorationCohortInterventionEvaluationWidget):
+    """
+    A widget for exploring the model performance of a cohort.
+    """
+
+    def __init__(self):
+        """
+        Exploration widget for viewing rates of interventions and outcomes accross categories in a cohort group.
+
+        Parameters
+        ----------
+        cohort_groups : dict[str, tuple[Any]]
+            cohort names and category values
+        outcome_names : tuple[Any], optional
+            outcome descriptors, by default None
+        intervention_names : tuple[Any], optional
+            intervention descriptors, by default None
+        reference_time_names : tuple[Any], optional
+            reference time descriptors, by default None
+        """
+        from seismometer.seismogram import Seismogram
+
+        sg = Seismogram()
+        reference_times = [sg.predict_time]
+        if sg.comparison_time and sg.comparison_time != sg.predict_time:
+            reference_times.append(sg.comparison_time)
+
+        super().__init__(
+            "Outcome / Intervention Analysis",
+            sg.available_cohort_groups,
+            tuple(sg.config.outcomes.keys()),
+            tuple(sg.config.interventions.keys()),
+            reference_times,
+        )
+
+    def generate_plot(self) -> HTML:
+        from seismometer._api import plot_intervention_outcome_timeseries
```

Comment:
So if we generalize that fn.__name__ bit, it feels like there is a way to shift the seam of controls to be only the options (plus plot signature).

I haven't looked completely but I'd hope we can move plotting imports completely out of the control submodule; instead _api can import controls and construct the callable instances having specified that last bit of info.

Reply:
FINC